test/tstDendrogram.cpp

```diff
@@ -188,4 +188,53 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(dendrogram_boruvka, DeviceType,
   BOOST_TEST(heights_boruvka == heights_union_find, tt::per_element());
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(dendrogram_boruvka_same_weights, DeviceType,
+                              ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  using MemorySpace = typename DeviceType::memory_space;
+
+  using namespace ArborX::Details;
+  using Point = ArborX::ExperimentalHyperGeometry::Point<2, float>;
+
+  ExecutionSpace space;
+
+  int const N = 101;
+  int const n = N * N;
+  std::vector<Point> points_v(N * N);
+  for (int i = 0; i < N; ++i)
+    for (int j = 0; j < N; ++j)
+      points_v[(2 * i % N) * N + (2 * j % N)] = Point{(float)i, (float)j};
```

Comment:
Doesn't that mean you don't initialize odd values here and overwrite even values?
Can you please add a comment explaining why/how you do the initialization here?